mat64/io.go

```diff
@@ -81,18 +87,24 @@ func (m *Dense) UnmarshalBinary(data []byte) error {
 		panic("mat64: unmarshal into non-zero matrix")
 	}
 
+	var tmp [8]byte
 	buf := bytes.NewReader(data)
-	var rows int64
-	err := binary.Read(buf, binary.LittleEndian, &rows)
+	_, err := buf.Read(tmp[:])
```

Comment:
Instead I'd use io.ReadFull here. It already does that check - returning io.ErrUnexpectedEOF. (same in other places where you're reading things that are not matrix elements).